mono/metadata/android-ifaddrs.c

```diff
@@ -0,0 +1,1158 @@
+					goto error;
+				}
+				ifa->ifa_addr = (struct sockaddr*)sa;
+				break;
+
+			default:
+				g_debug ("     rta_type: ");
+#if DEBUG
+				print_ifla_name (attribute->rta_type);
+#endif
+				break;
+		}
+
+		attribute = RTA_NEXT (attribute, length);
+	}
+	g_debug ("link flags: 0x%X", ifa->ifa_flags);
+	return ifa;
+
+  error:
+	if (sa)
+		free (sa);
+	free_single_xamarin_ifaddrs (&ifa);
+
+	return NULL;
+}
+
+#if DEBUG
+
+static void
+print_ifla_name (gint id)
+{
+	static struct {
+		gint   value;
+		gchar *name;
+	} iflas[] = {
+#define ENUM_VALUE_ENTRY(enumvalue) { enumvalue, #enumvalue }
+		ENUM_VALUE_ENTRY (IFLA_UNSPEC),
+		ENUM_VALUE_ENTRY (IFLA_ADDRESS),
+		ENUM_VALUE_ENTRY (IFLA_BROADCAST),
+		ENUM_VALUE_ENTRY (IFLA_IFNAME),
+		ENUM_VALUE_ENTRY (IFLA_MTU),
+		ENUM_VALUE_ENTRY (IFLA_LINK),
+		ENUM_VALUE_ENTRY (IFLA_QDISC),
+		ENUM_VALUE_ENTRY (IFLA_STATS),
+		ENUM_VALUE_ENTRY (IFLA_COST),
+		ENUM_VALUE_ENTRY (IFLA_PRIORITY),
+		ENUM_VALUE_ENTRY (IFLA_MASTER),
+		ENUM_VALUE_ENTRY (IFLA_WIRELESS),
+		ENUM_VALUE_ENTRY (IFLA_PROTINFO),
+		ENUM_VALUE_ENTRY (IFLA_TXQLEN),
+		ENUM_VALUE_ENTRY (IFLA_MAP),
+		ENUM_VALUE_ENTRY (IFLA_WEIGHT),
+		ENUM_VALUE_ENTRY (IFLA_OPERSTATE),
+		ENUM_VALUE_ENTRY (IFLA_LINKMODE),
+		ENUM_VALUE_ENTRY (IFLA_LINKINFO),
+		ENUM_VALUE_ENTRY (IFLA_NET_NS_PID),
+		ENUM_VALUE_ENTRY (IFLA_IFALIAS),
+		ENUM_VALUE_ENTRY (IFLA_NUM_VF),
+		ENUM_VALUE_ENTRY (IFLA_VFINFO_LIST),
+		ENUM_VALUE_ENTRY (IFLA_STATS64),
+		ENUM_VALUE_ENTRY (IFLA_VF_PORTS),
+		ENUM_VALUE_ENTRY (IFLA_PORT_SELF),
+		ENUM_VALUE_ENTRY (IFLA_AF_SPEC),
+		ENUM_VALUE_ENTRY (IFLA_GROUP),
+		ENUM_VALUE_ENTRY (IFLA_NET_NS_FD),
+		ENUM_VALUE_ENTRY (IFLA_EXT_MASK),
+		ENUM_VALUE_ENTRY (IFLA_PROMISCUITY),
+		ENUM_VALUE_ENTRY (IFLA_NUM_TX_QUEUES),
+		ENUM_VALUE_ENTRY (IFLA_NUM_RX_QUEUES),
+		ENUM_VALUE_ENTRY (IFLA_CARRIER),
+		ENUM_VALUE_ENTRY (IFLA_PHYS_PORT_ID),
+		ENUM_VALUE_ENTRY (IFLA_CARRIER_CHANGES),
+		ENUM_VALUE_ENTRY (IFLA_PHYS_SWITCH_ID),
+		ENUM_VALUE_ENTRY (IFLA_LINK_NETNSID),
+		ENUM_VALUE_ENTRY (IFLA_PHYS_PORT_NAME),
+		ENUM_VALUE_ENTRY (IFLA_PROTO_DOWN),
+		ENUM_VALUE_ENTRY (IFLA_GSO_MAX_SEGS),
+		ENUM_VALUE_ENTRY (IFLA_GSO_MAX_SIZE),
+		ENUM_VALUE_ENTRY (IFLA_PAD),
+		ENUM_VALUE_ENTRY (IFLA_XDP),
+		{ -1, 0 }
+#undef ENUM_VALUE_ENTRY
+	};
+
+	for (gint i = 0; iflas [i].value != -1 && iflas [i].name != 0; ++i) {
+		if (iflas [i].value == id) {
+			g_message ("ifla->name: %s (%d)\n", iflas [i].name, iflas [i].value);
+			return;
+		}
+	}
+
+	g_message ("Unknown ifla->name: unknown id %d\n", id);
+}
+
+static void
+print_address_list (gchar *title, struct _monodroid_ifaddrs *list)
+{
+	struct _monodroid_ifaddrs *cur;
+	gchar *msg, *tmp;
+
+	if (!list) {
+		g_message ("monodroid-net", "No list to print in %s", __FUNCTION__);
+		return;
+	}
+
+	cur = list;
+	msg = NULL;
+	while (cur) {
+		tmp = NULL;
+		asprintf (&tmp, "%s%s%p (%s; %p)", msg ? msg : "", msg ? " -> " : "", cur, cur->ifa_name, cur->ifa_name);
+		if (msg)
```

Comment:
No need for if.

Reply:
Once again, same comment here, tiny savings.

Reply:
Code bloat, wasting memory to optimize cycles for when low on memory.